src/main/java/uk/gov/justice/hmpps/prison/api/resource/BookingResource.java

```diff
@@ -298,8 +300,11 @@ public ResponseEntity<Void> updateAttendance(@PathVariable("offenderNo") @Parame
     @Operation(summary = "Update offender attendance and pay.", description = "Update offender attendance and pay.")
     @PutMapping("/{bookingId}/activities/{activityId}/attendance")
     @ProxyUser
-    public ResponseEntity<Void> updateAttendance(@NotNull @PathVariable("bookingId") @Parameter(description = "The booking Id of the prisoner", required = true, example = "213531") final Long bookingId, @NotNull @PathVariable("activityId") @Parameter(description = "The activity id", required = true, example = "1212131") final Long activityId, @RequestBody @NotNull @Parameter(required = true) final UpdateAttendance updateAttendance) {
-        bookingService.updateAttendance(bookingId, activityId, updateAttendance);
+    public ResponseEntity<Void> updateAttendance(@NotNull @PathVariable("bookingId") @Parameter(description = "The booking Id of the prisoner", required = true, example = "213531") final Long bookingId,
+                                                 @NotNull @PathVariable("activityId") @Parameter(description = "The activity id", required = true, example = "1212131") final Long activityId,
+                                                 @RequestParam(value = "lockTimeout", required = false) @Parameter(description = "Whether to timeout if locked", example = "true") final Boolean lockTimeout,
+                                                 @RequestBody @NotNull @Parameter(required = true) final UpdateAttendance updateAttendance) {
+        bookingService.updateAttendance(bookingId, activityId, updateAttendance, lockTimeout == Boolean.TRUE);
```

Comment:
very minor but i guess ` lockTimeout == Boolean.TRUE` can just be `lockTimeout `?

Reply:
I didnt check this, but i believe it is null when the param is not specified

Reply:
would be better to default it in the `RequestParam` instead e.g. https://github.com/ministryofjustice/prison-api/blob/669aafe910b68a677a1bd8df1c3979ef1748238d/src/main/java/uk/gov/justice/hmpps/prison/api/resource/PrisonerResource.kt#L51
```
    @RequestParam(value = "includeAliases", required = false, defaultValue = "false")
```